### DISCUSSION TITLE
GEN24_API

### DISCUSSION THREAD

**Discussion by Manniene** [wiggal/GEN24_Ladesteuerung#113] [Q&A]
Hi Wiggal, bekommst du über die API eigentlich auch ws/kWh Werte wie z.B. wieviel von der PV / Netz in die Batterie ging bzw. wieviel wieder rausgeht? Ich versuche den Wirkungsgrad für das Laden aus dem Netz in die Batterie  und dann wieder raus zu bestimmen. Die Daten aus dem Solarweb, Netzbezugsdaten über Stromzähler/Tibber und die Inverterdaten des GEN24 passen nicht verlässlich zusammen. Mit ein Grund ist, dass die Daten aus dem Solarweb sehr oft der falschen Stunde zugeordnet werden und dadurch teilweise sogar dem Folgetag zugeordnet werden. Der einzige Wert, der verlässlich zu sein scheint ist der Inverterwert direkt vom GEN24 (produzierte AC Menge). Dieser stimmt oft nicht mit der Summe aus (Direkt verbraucht + Aus Batterie bezogen) überein.

**Comment by wiggal** [comment]
Ja, die Werte, wie z.B. wie viel von der PV / Netz in die Batterie ging bzw. wie viel wieder herausgeht, sind in der API drin.
In der `http://XXX.XXX.XXX.XXX/components/readable`, hier mal die Quellen aus `FUNCTIONS/GEN24_API.py`:
```
API['AC_Produktion'] =  int(data['Body']['Data']['327680']['channels']['ACBRIDGE_ENERGYACTIVE_PRODUCED_SUM_U64']/3600)
API['DC_Produktion'] =  int((data['Body']['Data']['393216']['channels']['PV_ENERGYACTIVE_ACTIVE_SUM_01_U64']+ data['Body']['Data']['393216']['channels']['PV_ENERGYACTIVE_ACTIVE_SUM_02_U64'])/3600)
API['Batterie_IN'] =    ['393216']['channels']['BAT_ENERGYACTIVE_ACTIVECHARGE_SUM_01_U64']
API['Batterie_OUT'] =   ['393216']['channels']['BAT_ENERGYACTIVE_ACTIVEDISCHARGE_SUM_01_U64']
API['Netzverbrauch'] =  ['16252928']['channels']['SMARTMETER_ENERGYACTIVE_CONSUMED_SUM_F64'])
API['Einspeisung'] =    ['16252928']['channels']['SMARTMETER_ENERGYACTIVE_PRODUCED_SUM_F64'])
```
Dort kannst du auch mal hineinschauen, da sind noch mehr drin. Und die Zählerstände werden in der DB PV_Daten.sqlite gespeichert dort aber in Watt/h, die kannst du historisch auch anschauen (z.B. mit DB Browser for SQLite).

Schönen Gruß

**Reply by wiggal** [reply]
Das mit den Verlusten wird auch hier diskutiert:
https://github.com/evcc-io/evcc/discussions/16364#discussioncomment-11361575
